R/generate_eval_data.R

```diff
@@ -128,5 +141,28 @@ get_scores_for_output_type <- function(model_out_tbl, oracle_output, metric_name
     by = c("model_id", by)
   )
 
+  if (!is.null(relative_metrics)) {
+    # Return only the scaled relative metrics, not the unscaled ones
+    rel_skill_colnames <- paste0(relative_metrics, "_relative_skill")
+    scores <- dplyr::select(scores, !dplyr::all_of(rel_skill_colnames))
+
+    # Place scaled relative metric columns before corresponding metric columns
+    # relative_metrics is a subset of metrics
+    ordered_metric_cols <- purrr::map(
+      metrics,
+      function(metric) {
+        c(
+          if (metric %in% relative_metrics) paste0(metric, "_scaled_relative_skill") else NULL,
+          metric
+        )
+      }
+    ) |>
+      unlist()
```

Comment:
This is one of the downsides of purrr... you can't do `map_chr()` over more than one thing. Here's a vapply solution in five lines, but feel free to leave it. 

```suggestion
    add_skill <- function(m) c(paste0(m, "_scaled_relative_skill"), m)
    mmm_drop <- function(m) c("drop", m)
    res <- vapply(metrics, \(m) if (m %in% relative_metrics) add_skill(m) else mmm_drop(m), FUN.VALUE = character(2))
    ordered_metric_cols <- res[res != "drop"]
```